### PR TITLE
Expanded Multihost Test Infra

### PR DESCRIPTION
**Pull request by jameszianxuTT**
### Ticket
Partial resolution to #3127 

### Problem description
Currently there is very little multihost test infra, and none for a true multihost configuration (i.e. multiple physical machines).

### What's changed
- Add rank binding presets for quad_galaxy, dual_galaxy and dual_bh_quietbox
- Create unified multihost tests and test infra under tests/torch/multi_host/experimental/ to set TT_DISTRIBUTED_* env variables around tests
- Add remote_docker.sh script to demonstrate container to container ssh proxying 
- Add simple tests for multihost eltwise addition, contracting-dim sharded matmul requiring all reduce and sharded output generating batch sharded matmul requiring all gather
- Add gpt_oss 20B test to existing multihost on llmbox test infra
- Check in gpt_oss 120B from @ddilbazTT reference example for dual qb execution
- Check in llama 3.1 8B test for reference model on multigalaxy with test infra
- Flush loguru output buffer  

### Checklist
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/torch/multi_host/experimental/conftest.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    Args:
+        num_devices: Total number of devices across all hosts
+
+    Returns:
+        Tuple of (batch_dim, model_dim) for mesh shape
+
+    Examples:
+        8 devices -> (2, 4)   # dual_bh_quietbox
+        16 devices -> (2, 8)  # dual_galaxy
+        32 devices -> (4, 8)  # future topology
```
Comment (jameszianxuTT) [inline tests/torch/multi_host/experimental/conftest.py]: known typos in this comment

File: tests/torch/multi_host/experimental/test_gptoss-120b.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+import torch
+import torch_xla
+import torch_xla.distributed.spmd as xs
+import torch_xla.runtime as xr
+from torch_xla.distributed.spmd import Mesh
+from transformers import AutoConfig, AutoModelForCausalLM, AutoTokenizer
+from transformers.utils.quantization_config import Mxfp4Config
+
+
+def test_gpt_oss_120b():
```
[the remark this answers is not in the document]
Comment (AleksKnezevic) [reply tests/torch/multi_host/experimental/test_gptoss-120b.py]: this can be in a follow up PR if you want.

[the remark this answers is not in the document]
Comment (jameszianxuTT) [reply tests/torch/multi_host/experimental/test_gptoss-120b.py]: Okay, I think I'll do that. There will be more changes coming to integrate our tests into metal's existing multihost CI anyways, so I can fold this test_models infra integration there once I know what it'll look like.

**Review by vvukomanTT — APPROVED** [review]
